### PR TITLE
Added functional tests for Hello World and Kitchensink quickstarts

### PR DESCRIPTION
**Pull request by okiss**

### CODE REVIEW

**Comment by sgilda** [comment]
I ran the functional test for these 10 quickstarts: helloworld-errai, helloworld-gwt, helloworld-html5, kitchensink-angularjs, kitchensink-backbone, kitchensink-spring-asynrquestmapping, kitchensink-spring-basic, kitchensink-spring-controlleradvice, kitchensink-spring-matrixvariables, kitchensink-spring-springmvctest.

For kitchensink-spring-basic, kitchensink-spring-controlleradvice,and kitchensink-spring-springmvctest, the tests are successful but I do see the following warning at the end. Is that to be expected?

```
    [WARNING] The requested profile "arq-jbossas-remote" could not be activated because it does not exist
```

kitchensink-backbone test fails with: 

```
    Failed tests:   testRegister(org.jboss.as.quickstarts.kitchensink.test.MemberRegistrationTest): Unexpected response status expected:<200> but was:<409>
```

@joshuawilson or @pmuir: Do you need to do a code review?

**Comment by joshuawilson** [comment]
@sgilda I think I will need to review it.

**Comment by sgilda** [comment]
The warning appeared because I did not run the tests from the functional-tests diectory. Please ignore. Thanks!

**Comment by pmuir** [comment]
I get errors running the kitchensink-spring-asyncrequestmapping ftests:

```
Running org.jboss.as.quickstarts.kitchensink.test.KitchensinkSpringAsyncRequestMappingTest
Oct 10, 2013 5:08:40 PM org.jboss.arquillian.drone.webdriver.factory.remote.reusable.ReusedSessionPernamentFileStorage readStore
INFO: Reused session store is not available at /Users/pmuir/.drone-webdriver-session-store, a new one will be created.
Oct 10, 2013 5:08:41 PM org.jboss.arquillian.drone.impl.DroneEnhancer deenhanceDrone
WARNING: Deprecated Enhancer type, please implement DroneInstanceEnhancer instead. Deenhancing WebDriver @Default using enhancer org.jboss.arquillian.graphene.integration.GrapheneEnhancer with precedence -100
Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 1.872 sec <<< FAILURE!
Running org.jboss.as.quickstarts.kitchensink.test.RESTTest
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.027 sec <<< FAILURE!

Results :

Tests in error: 
  org.jboss.as.quickstarts.kitchensink.test.KitchensinkSpringAsyncRequestMappingTest: Could not invoke deployment method: public static org.jboss.shrinkwrap.api.spec.WebArchive org.jboss.as.quickstarts.kitchensink.test.KitchensinkSpringAsyncRequestMappingTest.deployment()
  org.jboss.as.quickstarts.kitchensink.test.KitchensinkSpringAsyncRequestMappingTest: Unexpected callable present in Drone Context, should be already instantiated at this moment.
  org.jboss.as.quickstarts.kitchensink.test.RESTTest: Could not invoke deployment method: public static org.jboss.shrinkwrap.api.spec.WebArchive org.jboss.as.quickstarts.kitchensink.test.RESTTest.deployment()

Tests run: 3, Failures: 0, Errors: 3, Skipped: 0
```

**Comment by pmuir** [comment]
I get errors running kitchensink-spring-controlleradvice:

```
Running org.jboss.as.quickstarts.kitchensink.test.KitchensinkSpringControllerAdviceTest
Oct 10, 2013 5:15:41 PM org.jboss.arquillian.drone.webdriver.factory.remote.reusable.ReusedSessionPernamentFileStorage readStore
INFO: Reused session store is not available at /Users/pmuir/.drone-webdriver-session-store, a new one will be created.
Oct 10, 2013 5:15:57 PM org.jboss.arquillian.drone.impl.DroneEnhancer enhanceDrone
WARNING: Deprecated Enhancer type, please implement DroneInstanceEnhancer instead. Enhancing WebDriver @Default using enhancer org.jboss.arquillian.graphene.integration.GrapheneEnhancer with precedence -100
Oct 10, 2013 5:16:12 PM org.jboss.arquillian.drone.impl.DroneEnhancer deenhanceDrone
WARNING: Deprecated Enhancer type, please implement DroneInstanceEnhancer instead. Deenhancing WebDriver @Default using enhancer org.jboss.arquillian.graphene.integration.GrapheneEnhancer with precedence -100
Tests run: 6, Failures: 0, Errors: 4, Skipped: 0, Time elapsed: 32.603 sec <<< FAILURE!
Running org.jboss.as.quickstarts.kitchensink.test.RESTTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.235 sec

Results :

Tests in error: 
  testRegistrationWithBadNameFormat(org.jboss.as.quickstarts.kitchensink.test.KitchensinkSpringControllerAdviceTest): Cannot locate element using: a.ui-state-highlight
  testRegistrationWithBadEmailFormat(org.jboss.as.quickstarts.kitchensink.test.KitchensinkSpringControllerAdviceTest): Cannot locate element using: a.ui-state-highlight
  testRegistrationWithBadPhoneFormat(org.jboss.as.quickstarts.kitchensink.test.KitchensinkSpringControllerAdviceTest): Cannot locate element using: a.ui-state-highlight
  testRegularRegistration(org.jboss.as.quickstarts.kitchensink.test.KitchensinkSpringControllerAdviceTest): Cannot locate element using: a.ui-state-highlight

Tests run: 7, Failures: 0, Errors: 4, Skipped: 0
```

**Comment by pmuir** [comment]
I fixed the version issues on a branch - https://github.com/pmuir/jboss-wfk-quickstarts/tree/ftests - and code looks good. But we have the issues I identified above to resolve before we merge.

@joshuawilson these are our own quickstarts right? not copied from spring.

**Comment by joshuawilson** [comment]
@pmuir yes these are our's so we can change them as much as needed or desired.  They are just meant to be helpful.

File: helloworld-errai/functional-tests/pom.xml [hunk 1]
```diff
@@ -0,0 +1,131 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rafabene) [inline helloworld-errai/functional-tests/pom.xml]: Doesn't these [functional-tests] added as a maven submodule in the parent pom.xml ?

**Pull request merged**